src/main/scala/viper/gobra/translator/util/TypePatterns.scala

```diff
@@ -192,11 +192,31 @@ object TypePatterns {
 
     object Struct {
       def unapply(arg: in.Type): Option[Vector[in.Field]] = underlyingType(arg)(ctx) match {
-        case t : in.StructT => Some(t.fields)
+        case t: in.StructT => Some(t.fields.filter(!_.notImported))
         case _ => None
       }
     }
 
+    /* imported struct with private fields */
+    object PartialStruct {
+      def unapply(arg: in.Type): Option[Vector[in.Field]] = underlyingType(arg)(ctx) match {
+        case t: in.StructT =>
+          val existingFields = t.fields.filter(!_.notImported)
+          if (existingFields.size == t.fields.size) None
+          else Some(existingFields)
+        case _ => None
+      }
+    }
+
+    object CompleteStruct {
+      def unapply(arg: in.Type): Option[Vector[in.Field]] = underlyingType(arg)(ctx) match {
+        case t: in.StructT =>
+          if (t.fields.exists(_.notImported)) None
+          else Some(t.fields)
+        case _ => None
+      }
+    } 
```

Comment:
I thought about whether this check is sufficient to then choose different encodings accordingly. I think it is not enough. Consider a type `struct Pair { Left, Right int}`. Even though the struct does not have private fields, it has to be encoded with the partial tuple encoding in client code because clients must not be able to deduce facts about equality, e.g. that `x.Left == y.Left && x.Right == y.Right` must not imply `x == y`.

Reply:
Do you mean for code like this:
```
x := bar.Pair{Left: 2, Right: 0} 
y := bar.Pair{Left: 2, Right: 0}
assert(x == y) //does not fail
```

It already works for:
```
x2 := &bar.Pair{Left: 2, Right: 0}
y2 := &bar.Pair{Left: 2, Right: 0}

//:: ExpectedOutput(error:assertion_error)
assert(x2 == y2) //fails
```

Reply:
I had the following code in mind:

```go=
func foo(x, y bar.Pair) {
  if x.Left == y.Left && x.Right == y.Right {
    assert x == y
  }
}
```

Your first example should verify because literal creation is deterministic. The second example should always fail because you are comparing pointers to different memory locations.